autobahn/wamp/interfaces.py

```diff
@@ -672,6 +679,13 @@ def subscribe(self, handler, topic=None, options=None):
         :param options: Options for subscribing.
         :type options: An instance of :class:`autobahn.wamp.types.SubscribeOptions`.
 
+        :param check_types: Enable automatic type checking against (Python 3.5+) type hints
+            specified on the ``endpoint`` callable. Types are checked at run-time on each
+            invocation of the ``endpoint`` callable. When a type mismatch occurs,
+            an :class:`autobahn.wamp.error.TypeError` is raised automatically and returned
+            to the callee (via the router).
```

Comment:
rethinking this, we should probabyl:

* for the RPC case (register): return the error to the _caller_ and additionally raise the error in `onUserError` (within the _callee_ side)
* for the PubSub case (subscribe): only raise the error in `onUserError` (within the _subscriber_ side, as there is no message back to the _publisher_ nor to the router - until we have "acknowledged event delivery" in WAMP)

Reply:
Latest commit tries to address that, kindly take a look.

Reply:
Improved the docstrings a bit...